cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
+	if err != nil {
+		return nil, nil, err
+	}
+
+	orgSvc := &tenant.OrgClientService{Client: httpClient}
+	return &http.TaskService{Client: httpClient}, orgSvc, nil
+}
+
 func cmdTask(f *globalFlags, opt genericCLIOpts) *cobra.Command {
-	runE := func(cmd *cobra.Command, args []string) error {
-		seeHelp(cmd, args)
-		return nil
+	builder := newCmdTaskBuilder(newTaskSVCs, f, opt)
+	return builder.cmd()
+}
+
+type cmdTaskBuilder struct {
+	opts        genericCLIOpts
+	globalFlags *globalFlags
+
+	svcFn taskSVCsFn
+
+	taskID         string
+	runID          string
+	taskPrintFlags taskPrintFlags
+	// todo: fields of these flags structs could be pulled out for a more streamlined builder struct
+	taskCreateFlags      taskCreateFlags
+	taskFindFlags        taskFindFlags
+	taskRerunFailedFlags taskRerunFailedFlags
+	taskUpdateFlags      taskUpdateFlags
+	taskRunFindFlags     taskRunFindFlags
+	org                  organization
+}
+
+func newCmdTaskBuilder(svcsFn taskSVCsFn, f *globalFlags, opts genericCLIOpts) *cmdTaskBuilder {
+	return &cmdTaskBuilder{
+		globalFlags: f,
+		opts:        opts,
+		svcFn:       svcsFn,
 	}
+}
 
-	cmd := opt.newCmd("task", runE, false)
+func (t *cmdTaskBuilder) cmd() *cobra.Command {
+	cmd := t.newCmd("task", nil)
 	cmd.Short = "Task management commands"
-
+	cmd.TraverseChildren = true
+	cmd.Run = seeHelp
 	cmd.AddCommand(
-		taskLogCmd(f, opt),
-		taskRunCmd(f, opt),
-		taskCreateCmd(f, opt),
-		taskDeleteCmd(f, opt),
-		taskFindCmd(f, opt),
-		taskUpdateCmd(f, opt),
+		t.taskLogCmd(),
+		t.taskRunCmd(),
+		t.taskCreateCmd(),
+		t.taskDeleteCmd(),
+		t.taskFindCmd(),
+		t.taskUpdateCmd(),
+		t.taskRerunFailedCmd(),
 	)
 
 	return cmd
 }
 
-var taskPrintFlags struct {
+func (t *cmdTaskBuilder) newCmd(use string, runE func(*cobra.Command, []string) error) *cobra.Command {
+	cmd := t.opts.newCmd(use, runE, true)
+	t.globalFlags.registerFlags(t.opts.viper, cmd)
+	return cmd
+}
+
+type taskPrintFlags struct {
 	json        bool
 	hideHeaders bool
 }
 
-var taskCreateFlags struct {
-	org  organization
+type taskCreateFlags struct {
 	file string
 }
 
-func taskCreateCmd(f *globalFlags, opt genericCLIOpts) *cobra.Command {
-	cmd := opt.newCmd("create [script literal or -f /path/to/script.flux]", taskCreateF, true)
+func (t *cmdTaskBuilder) taskCreateCmd() *cobra.Command {
+	cmd := t.newCmd("create [script literal or -f /path/to/script.flux]", t.taskCreateF)
 	cmd.Args = cobra.MaximumNArgs(1)
 	cmd.Short = "Create task"
 	cmd.Long = `Create a task with a Flux script provided via the first argument or a file or stdin`
 
-	f.registerFlags(opt.viper, cmd)
-	cmd.Flags().StringVarP(&taskCreateFlags.file, "file", "f", "", "Path to Flux script file")
-	taskCreateFlags.org.register(opt.viper, cmd, false)
-	registerPrintOptions(opt.viper, cmd, &taskPrintFlags.hideHeaders, &taskPrintFlags.json)
+	cmd.Flags().StringVarP(&t.taskCreateFlags.file, "file", "f", "", "Path to Flux script file")
+	t.org.register(t.opts.viper, cmd, false)
```

Comment:
Looks like this is repeated in all the commands. Is it possible to call once as part of `(t *cmdTaskBuilder).cmd()`?